File: PHP/pinput.php
<?php

function ParseInput($argv, $argc, $flags){
	$res = array();
	$res['parameter'] = array();
	$res['optional'] = array();
	$res['flag'] = array();


	$optionalParameterKey = '';
	$isOptionalParameter = false;

	for($i = 1; $i < $argc; ++$i){
		$arg = $argv[$i];
		if($arg[0] == '-'){
			$isOptionalParameter = false;
			if(strlen($arg) > 1){
				if($arg[1] == '-' || strlen($arg) == 2){
					if(isset($flags[$arg])){
						$res['flag'][$arg] = 1;
					}
					else{
						$res['optional'][$arg] = array();
						$optionalParameterKey = $arg;
						$isOptionalParameter = true;
					}
				}
				else{
					for($j = 1; $j < strlen($arg); ++$j){
						$res['flag']['-'.$arg[$j]] = 1;
					}
				}
			}
		}
		else if($isOptionalParameter){
			array_push($res['optional'][$optionalParameterKey],$arg);
		}
		else{
			array_push($res['parameter'],$arg);
		}
	}

	return $res;
}
?>

File: python/pinput.py
from sets import Set

def parse(args, flagsSet, optionalParametersSet):

	parameters = []
	optionalParameters = {}
	flags = Set([])


	optionalParameterKey = ''
	isOptionalParameter = False

	for i in range(1,len(args)):
		arg = args[i]
		if arg[0] == '-':
			isOptionalParameter = False
			if len(arg) > 1:
				if arg[1] == '-' or len(arg) == 2:
					if arg in flagsSet:
						flags.add(arg)
					else:
						optionalParameters[arg] = []
						optionalParameterKey = arg
						isOptionalParameter = True
				else:
					for j in range(1,len(arg)):
						flags.add('-'+arg[j])

		elif isOptionalParameter:
			optionalParameters[optionalParameterKey] += [arg]
		else:
			parameters += [arg]

	return parameters, optionalParameters, flags